app/controllers/spree/shipwire_webhook_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Spree::ShipwireWebhookController < ActionController::Base
+  respond_to :json
+
+  before_action :validate_key, except: :subscribe
```

Comment:
This controller can be extended by others controller. All actions need to be validated before execute